handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -907,9 +914,9 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             // See https://github.com/netty/netty/issues/1534
 
             in.skipBytes(totalLength);
-            final ByteBuffer inNetBuf = in.nioBuffer(startOffset, totalLength);
+            final ByteBuf inNetBuf = in.slice(startOffset, totalLength);
             unwrap(ctx, inNetBuf, totalLength);
```

Comment:
Instead of creating a new slice, please modify `unwrap(ctx, packet, length)` so that it accepts the case where the reader index of the packet is not zero.